src/client.ts

```diff
@@ -216,7 +220,21 @@ export default class ScubaClient {
         const resp = (await this._api.getMetricsBatch(metricsClass, body, {
             ...this._defaultReqOptions,
             ...options,
-        })) as any;
+        })) as unknown as { data: GetMetricsBatchResponse };
+        resp.data.metrics.forEach(resource => {
+            if ('error' in resource) {
+                return;
+            }
+            resource.metrics?.forEach(metric => {
+                if ('error' in metric) {
+                    return;
+                }
+                // eslint-disable-next-line no-param-reassign
+                metric.bytesTotal = BigInt(metric.bytesTotal || 0);
```

Comment:
if it were (untyped) javascript, this would be consistent; however, here we kind of break/abuse the typing system : at line 223, we say the response contains a metrics with `bigint` parameters.... so this line would be (according to the type-system) useless

* Is typescript allowed to optimize away such seamlingly useless conversions? If it is, we have an issue
* The cleaner way may be to have 2 types (one "internal" with strings, and another for the API) ; or to tweak the JSON deserialization, maybe with a "[reviver](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#using_the_reviver_parameter)" (c.f. https://stackoverflow.com/a/69644630)

Reply:
To me, it's more that we use the same `GetMetricsBatchResponse` type for both the received response and the function return type, while it should be two different types: we receive strings, and we return bigints.

I am against revivers, it slows down serialization and may lead to problem if we were to rely on it and eventually get BigInt in places we are not expecting them (because if we use revivers here, we should to it everywhere).

Reply:
As long as we control the objects we get, there is no point using reviver, better convert it explicitly when needed. I will update the types to differentiate between the received string and the returned bigint.

Reply:
Done.